### PR TITLE
Email dispatcher if a mobile worker is in a predefined area

### PR DESCRIPTION
**Pull request by niravi**
Very basic geofencing; wanted to play around with some Python SMTP libraries so this was the result. I have a test account and data ready for this - we can discuss during our meeting tomorrow if we want to work on this more

### CODE REVIEW

File: scripts/geofence_email_notifications.py [hunk 1]
```diff
@@ -0,0 +1,128 @@
...
+   limitations under the License.​
+
+"""
+import argparse
+import logging
+import logging.handlers
+import traceback
+import pendulum
+import sys
+import yagmail
```
Comment (apulverizer) [inline scripts/geofence_email_notifications.py:20]: I would prefer we not use a third party library and just use the standard library, it seems simple enough.

File: scripts/geofence_email_notifications.py [hunk 2]
```diff
@@ -0,0 +1,128 @@
...
+        sys.exit(0)
+    
+    logger.info("Buffering around the provided lat/long ")
+    pt = Point({"x": args.longitude, "y": args.latitude, "spatialReference": {"wkid": 4326}})
+    buffer = arcgis.geometry.buffer([pt], distances=args.distance_tolerance, in_sr=4326, out_sr=4326, unit=9001)
+    buffer_filter = arcgis.geometry.filters.intersects(buffer[0], sr=4326)
+    results = lkl_layer.query(geometry_filter=buffer_filter).features
+    if len(results) > 0:
+        # ensure yagmail is set up before you hit this part of the code: https://github.com/kootenpv/yagmail#username-and-password
+        yag = yagmail.SMTP(args.gmail_user)
```
Comment (apulverizer) [inline scripts/geofence_email_notifications.py:87]: I would split this into it's own function like `send_email(...)`

File: scripts/geofence_email_notifications.py [hunk 3]
```diff
@@ -0,0 +1,128 @@
...
+if __name__ == "__main__":
+    # Get all of the commandline arguments
+    parser = argparse.ArgumentParser("Email if a Tracker user is in a geofence")
+    parser.add_argument('-u', dest='username', help="The username to authenticate with", required=True)
+    parser.add_argument('-p', dest='password', help="The password to authenticate with", required=True)
+    parser.add_argument('-org', dest='org_url', help="The url of the org/portal to use", required=True)
+    # Parameters for tracker
+    parser.add_argument('-lkl-layer-url', dest='lkl_layer_url', required=True,
+                        help="The last known location (LKL) layer (either location tracking service or tracks view) you'd like to use. This URL should end in /1")
+    parser.add_argument('-latitude', dest='latitude',type=float,required=True, help="Latitude of the point you want to buffer")
```
Comment (apulverizer) [inline scripts/geofence_email_notifications.py:111]: I think ideally the input would be a feature layer of polygons. There might be a `description` field the user can specify that could be used in the message.

**Review by apulverizer — COMMENTED** [review]
Let's chat more about this.

**Comment by niravi** [comment]
Closing in favor of https://github.com/Esri/workforce-scripts/pull/57

**Pull request closed without merging**